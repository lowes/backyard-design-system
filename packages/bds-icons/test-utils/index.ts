import { customRender } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }
