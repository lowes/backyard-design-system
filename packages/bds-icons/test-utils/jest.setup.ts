import '@testing-library/jest-dom'
import 'jest-styled-components'

jest.setTimeout(1000 * 60 * 20)
