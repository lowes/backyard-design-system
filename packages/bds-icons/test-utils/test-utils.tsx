import * as React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import * as theme from '@lowes-tech/bds-tokens/v2/light/theme'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
const customRender = (ui, options) => render(ui, { wrapper: RenderWrapper, ...options })

export { customRender }
