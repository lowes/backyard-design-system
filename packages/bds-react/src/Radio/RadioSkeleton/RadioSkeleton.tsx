import * as React from 'react'
import classnames from 'classnames'
import type { BDSForwardRef } from '../../utils/typings/BDSComponentProps'

import Skeleton from '../../Skeleton'
import SkeletonWrapper from './styles'

/**
 * Backard RadioSkeleton component.
 * 
 * Returns a skeleton component with a default width of 300px and a default
 * quantity of list items set to 4. The density is also set to normal by 
 * default.
 * 
 * <RadioSkeleton />
*/

const RadioSkeleton: BDSForwardRef<RadioSkeletonProps> = React.forwardRef<
    RadioSkeletonRef,
    RadioSkeletonProps
>(function RadioSkeleton (
    {
        animate = false,
        labelWidth = 'size_84',
        helperWidth = 'size_84',
        withLabel = false,
        withHelper = false,
        className,
        ...props
    },
    ref,
) {

    return (
        <SkeletonWrapper clasName={classnames(className, 'radio--skeleton')} {...props} ref={ref}>
            <div className='label-wrapper'>
                <Skeleton 
                    variant='circle'
                    height='size_20'
                    width='size_20'
                    animate={animate}
                />
                {
                    (withLabel) ?
                        <Skeleton
                            variant='rect'
                            height='size_20'
                            width={labelWidth}
                            animate={animate}
                            className='label--skeleton'
                        /> : null
                }
            </div>
            {
                (withHelper) ?
                    <Skeleton
                        variant='rect'
                        height='size_16'
                        width={helperWidth}
                        className='helper-text--skeleton'
                        animate={animate}
                    /> : null
            }
        </SkeletonWrapper>
    )
})

type RadioSkeletonRef = HTMLDivElement

interface RadioSkeletonProps {
    animate?: boolean
    withLabel?: boolean
    withHelper?: boolean
    labelWidth?: string
    helperWidth?: string
    className?: string
}

RadioSkeleton.bdsName = 'RadioSkeleton'

export { RadioSkeleton }

export type { RadioSkeletonProps, RadioSkeletonRef }

export default RadioSkeleton