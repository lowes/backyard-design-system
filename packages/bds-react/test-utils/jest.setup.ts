import * as React from 'react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

jest.setTimeout(1000 * 60 * 20)

// Mock `useRandomId` hook
// gives consistent ids instead
// of randomly generated ones for
// snapshot tests
let mockId = 0

jest.mock('../src/utils/hooks/useRandomId', () => {
    return jest.fn((id) => id || `${mockId++}`)
})

// Before each test...
beforeEach(() => {
    // Reset `useRandomId` counter
    mockId = 0
})
