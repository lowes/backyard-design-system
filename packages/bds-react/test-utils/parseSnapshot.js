const parseSnapshot = (content) => {
    const data = get_data(content)

    return Object.keys(data).reduce(
        (parsed, key) => ({ ...parsed, [key]: parse(data[key]) }),
        {},
    )
}

function parse(value) {
    const reverted_value = revert_extra_line_breaks(value).replace(/\n/g, '\\n')

    try {
        return reverted_value === 'undefined'
            ? undefined
            : JSON.parse(reverted_value)
    } catch (error) {
        // istanbul ignore next
        throw new Error(
            `Unexpected error while parsing:\n\n>>> start >>>\n${reverted_value}\n<<< end <<<\n\n${
            error.message
            }`,
        )
    }
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L128
function revert_extra_line_breaks(str) {
    return str.includes('\n') ? str.slice(1, -1) : str
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L98
function get_data(content) {
    const data = Object.create(null)
    const populate = new Function('exports', content)
    
    populate(data)
    
    return data
}

export {
    parseSnapshot
}

export default parseSnapshot
