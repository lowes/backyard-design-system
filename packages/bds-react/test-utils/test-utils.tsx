import * as React from 'react'
import { render } from '@testing-library/react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/ThemeProvider'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme="light">{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
// @ts-ignore
const customRender = (ui, options = {}) => render(ui, { wrapper: RenderWrapper, ...options })

/**
 * Function to override custom mount with
 *
 * @param {*} ui
 * @param {*} options
 */
const customMount = (ui, options = {}) =>
    mount(ui, {
        wrappingComponent: ThemeProvider,
        wrappingComponentProps: { theme: 'light' },
        ...options,
    })

export { customRender, customMount }
