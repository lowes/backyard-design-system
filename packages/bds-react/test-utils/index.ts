// @ts-ignore
import { customRender, customMount } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }

// Override shallow method
export { customMount as mount }
