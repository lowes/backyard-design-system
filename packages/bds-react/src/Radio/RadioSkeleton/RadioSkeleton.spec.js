import * as React from 'react'
import { render } from '../../../test-utils'
import RadioSkeleton from './RadioSkeleton'

describe('RadioSkeleton Snapshots', () => {
    test('RadioSkeleton default snapshot', () => {
        const { asFragment } = render(
            <RadioSkeleton />
        )

        expect(asFragment()).toMatchSnapshot()
    })


    test('RadioSkeleton animate snapshot', () => {
        const { asFragment } = render(
            <RadioSkeleton animate={true} />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('RadioSkeleton with label snapshot', () => {
        const { asFragment } = render(
            <RadioSkeleton withLabel labelWidth='8rem' />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('RadioSkeleton with label snapshot', () => {
        const { asFragment } = render(
            <RadioSkeleton withLabel labelWidth='8rem' withHelper helperWidth='6rem' />
        )

        expect(asFragment()).toMatchSnapshot()
    })
})