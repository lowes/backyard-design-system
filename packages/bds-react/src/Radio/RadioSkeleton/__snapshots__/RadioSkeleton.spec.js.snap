// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioSkeleton Snapshots RadioSkeleton animate snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1.25rem;
  height: 1.25rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1.25rem;
  height: 1.25rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .label-wrapper {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 .label-wrapper .label--skeleton {
  margin-left: 1rem;
}

.c0 .helper-text--skeleton {
  margin-top: 0.5rem;
  margin-left: 2.25rem;
}

<div
    class="c0"
  >
    <div
      class="label-wrapper"
    >
      <div
        class="c1 variant--circle shape--rounded style--animated"
        height="size_20"
        width="size_20"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`RadioSkeleton Snapshots RadioSkeleton default snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1.25rem;
  height: 1.25rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1.25rem;
  height: 1.25rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .label-wrapper {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 .label-wrapper .label--skeleton {
  margin-left: 1rem;
}

.c0 .helper-text--skeleton {
  margin-top: 0.5rem;
  margin-left: 2.25rem;
}

<div
    class="c0"
  >
    <div
      class="label-wrapper"
    >
      <div
        class="c1 variant--circle shape--rounded style--default"
        height="size_20"
        width="size_20"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`RadioSkeleton Snapshots RadioSkeleton with label snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1.25rem;
  height: 1.25rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1.25rem;
  height: 1.25rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c2 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c2.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c2.variant--circle {
  width: 8rem;
  height: 1.25rem;
  border-radius: var(--bds-border-radius-circle);
}

.c2.variant--rect {
  width: 8rem;
  height: 1.25rem;
}

.c2.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--rect.shape--squared {
  border-radius: 0;
}

.c2.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c2.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .label-wrapper {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 .label-wrapper .label--skeleton {
  margin-left: 1rem;
}

.c0 .helper-text--skeleton {
  margin-top: 0.5rem;
  margin-left: 2.25rem;
}

<div
    class="c0"
  >
    <div
      class="label-wrapper"
    >
      <div
        class="c1 variant--circle shape--rounded style--default"
        height="size_20"
        width="size_20"
      />
      <div
        class="c2 variant--rect shape--rounded label--skeleton style--default"
        height="size_20"
        width="8rem"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`RadioSkeleton Snapshots RadioSkeleton with label snapshot 2`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 1.25rem;
  height: 1.25rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 1.25rem;
  height: 1.25rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c2 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c2.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c2.variant--circle {
  width: 8rem;
  height: 1.25rem;
  border-radius: var(--bds-border-radius-circle);
}

.c2.variant--rect {
  width: 8rem;
  height: 1.25rem;
}

.c2.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--rect.shape--squared {
  border-radius: 0;
}

.c2.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c2.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--text.shape--squared {
  border-radius: 0;
}

.c3 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c3.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c3.variant--circle {
  width: 6rem;
  height: 1rem;
  border-radius: var(--bds-border-radius-circle);
}

.c3.variant--rect {
  width: 6rem;
  height: 1rem;
}

.c3.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c3.variant--rect.shape--squared {
  border-radius: 0;
}

.c3.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c3.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c3.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .label-wrapper {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 .label-wrapper .label--skeleton {
  margin-left: 1rem;
}

.c0 .helper-text--skeleton {
  margin-top: 0.5rem;
  margin-left: 2.25rem;
}

<div
    class="c0"
  >
    <div
      class="label-wrapper"
    >
      <div
        class="c1 variant--circle shape--rounded style--default"
        height="size_20"
        width="size_20"
      />
      <div
        class="c2 variant--rect shape--rounded label--skeleton style--default"
        height="size_20"
        width="8rem"
      />
    </div>
    <div
      class="c3 variant--rect shape--rounded helper-text--skeleton style--default"
      height="size_16"
      width="6rem"
    />
  </div>
</DocumentFragment>
`;
